rust/etw_helpers/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+#[cfg(any(feature = "crossbeam"))]
+pub mod event_consumer_crossbeam;
+#[cfg(any(feature = "flume"))]
+pub mod event_consumer_flume;
+#[cfg(any(feature = "futures"))]
+mod event_consumer_futures;
+#[cfg(any(feature = "futures"))]
+mod event_consumer_futures_stream;
+//mod event_consumer_raw;
+mod error;
+mod event_record;
+mod processtrace;
+mod sessions;
+
+/// If only one of the event consumer features are enabled,
+/// this module will be enabled as well to provide a generic naming scheme for accessing it.
+/// This allows for the selected feature to be switched without requiring code changes.
+///
```

Comment:
I feel like this is too clever, and won't be useful in practice. It only works if consumer X can force etw_helpers to only have one feature enabled, but cargo is explicitly designed to not allow a consumer to force a feature to be disabled -- a consumer can only force a feature to be enabled.